scipy/signal/_signaltools.py

```diff
@@ -3957,8 +3982,13 @@ def vectorstrength(events, period):
         fixed.  Biol Cybern. 2013 Aug;107(4):491-94.
         :doi:`10.1007/s00422-013-0560-8`.
     '''
-    events = np.asarray(events)
-    period = np.asarray(period)
+    xp = array_namespace(events, period)
+
+    events = xp.asarray(events)
+    period = xp.asarray(period)
+    if xp.isdtype(period.dtype, 'integral'):
+        period = xp.astype(period, xp.float64)
```

Comment:
Is it correct to jump straight to `float64`, what if events was `float32`? I think it would be better to use https://github.com/scipy/scipy/blob/1a3c042112d7d7a1e0959ac5039c6bb409db3dad/scipy/_lib/_array_api.py#L572
With `force_floating=True`

Reply:
This is an enhancement request, correct? Previously it was

```
In [1]: import numpy as np

In [2]: events = np.arange(4, dtype=np.float32)

In [3]: period = 3    # integer

In [4]: events, period = np.atleast_2d(events, period)

In [5]: np.exp(np.dot(2j*np.pi/period.T, events)).dtype
Out[5]: dtype('complex128')
```

Reply:
Ah ok thanks for confirming, I am on mobile so couldn't just try